src/Files.App/Utils/Cloud/Detector/GoogleDriveCloudDetector.cs

```diff
@@ -89,6 +120,218 @@ await FilesystemTasks.Wrap(() => StorageFile.GetFileFromPathAsync(Path.Combine(a
 					IconData = iconFile is not null ? await iconFile.ToByteArrayAsync() : null,
 				};
 			}
+
+#if DEBUG
+			await Inspect(database, "SELECT * FROM roots", "root_preferences db, roots table");
+			await Inspect(database, "SELECT * FROM media", "root_preferences db, media table");
+			await Inspect(database, "SELECT name FROM sqlite_master WHERE type = 'table' ORDER BY 1",
+				"root_preferences db, all tables");
+#endif
+
+			await foreach (var provider in GetGoogleDriveProvidersFromRegistryAsync())
+			{
+
+#if DEBUG
+				Debug.WriteLine("YIELD RETURNING from GoogleDriveCloudDetector#GetProviders (registry): ");
+				Debug.WriteLine("name=" + provider.Name + "; path=" + provider.SyncFolder);
+#endif
+
+				yield return provider;
+			}
+		}
+
+		private async Task Inspect(SqliteConnection database, string sqlCommand, string contentsOf)
+		{
+			await using var cmdTablesAll =
+				new SqliteCommand(sqlCommand, database);
+			var reader = await cmdTablesAll.ExecuteReaderAsync();
+			var colNamesList = Enumerable.Range(0, reader.FieldCount).Select(j => reader.GetName(j)).ToList();
+			var i = 0;
+			Debug.WriteLine("BEGIN LOGGING of " + contentsOf + " contents");
+			while (reader.Read())
+			{
+				var colVals = new object[reader.FieldCount];
+				reader.GetValues(colVals);
+				colVals.Select((val, j) => $"row {i}: column {j}: {colNamesList[j]}: {val}").ToList()
+					.ForEach(s => Debug.WriteLine(s));
+				i++;
+			}
```

Comment:
```suggestion

#if DEBUG
			Debug.WriteLine($"BEGIN LOGGING of {contentsOf} contents");
#ifdef

			if (int index = 0; reader.Read() is not false; index++)
			{
				var colVals = new object[reader.FieldCount];
				reader.GetValues(colVals);

				colVals.Select((val, j) => $"row {index}: column {j}: {colNamesList[j]}: {val}")
					.ToList().ForEach(s => Debug.WriteLine(s));
			}
```

Reply:
This is missing.

Reply:
Thanks, somehow I missed the loop change. I just pushed a commit that changes it to a `for` loop.